### PR TITLE
two background gfx functions

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀 This PR matches 2 functions (+0.24%) on `us`.

⚠️ This PR introduces 2 warnings:
- include/variables.h:272: warning: redundant redeclaration of `gPartnerActionStatus' in same scope
- include/variables.h:273: warning: redundant redeclaration of `gUIStatus' in same scope
- include/variables.h:274: warning: redundant redeclaration of `gPlayerStatus' in same scope
- include/variables.h:275: warning: redundant redeclaration of `gPlayerSpinState' in same scope
- include/variables.h:276: warning: redundant redeclaration of `gPlayerData' in same scope
- src/16F740.c:11: warning: redundant redeclaration of `gBackgroundImage' in same scope

**Review by ethteck — APPROVED** [review]

**Comment by ethteck** [comment]
some / all of those warnings are kinda unavoidable until we make some changes to BSS handling, so I wouldn't worry too much about them. great job on these

**Comment by Unnunu** [comment]
anyway i can fix the last warning

File: src/16c8e0.c [hunk 1]
```diff
@@ -713,17 +713,142 @@ void func_8023ED5C(void) {
...
+        fogEnabled = TRUE;
+        get_world_fog_color(&fogR, &fogG, &fogB, &fogA);
+        fogA = gGameStatusPtr->backgroundDarkness;
+        get_background_color_blend(&r1, &g1, &b1, &a1);
+        if (fogA == 255) {
+            for (i = 0; i < 256; i++) {
+                gTattleBgPalette[i] = 1;
+            }
+        } else {
+            for (i = 0; i < 256; i++) {
```
Comment (nanaian) [inline src/16c8e0.c:747]: Use `ARRAY_COUNT(gGameStatusPtr->backgroundPalette)`

File: src/16c8e0.c [hunk 2]
```diff
@@ -713,17 +713,142 @@ void func_8023ED5C(void) {
...
+                u16 palColor = ((u16*)gGameStatusPtr->backgroundPalette)[i];
```
Comment (nanaian) [inline src/16c8e0.c:748]: Is this cast required?

**Review by nanaian — CHANGES_REQUESTED** [review]

**Pull request merged**